spec.html

```diff
@@ -8854,13 +8854,15 @@ <h1>CreateBuiltinFunction ( _steps_, _internalSlotsList_ [ , _realm_ [ , _protot
     </emu-clause>
 
     <emu-clause id="sec-createbuiltingeneratorinstance" aoid="CreateBuiltinGeneratorInstance">
-      <h1>CreateBuiltinGeneratorInstance ( _closure_ [ , _generatorBrand_ [ , _generatorPrototype_ ] ] )</h1>
-      <p>The abstract operation CreateBuiltinGeneratorInstance takes arguments _closure_ (an Abstract Closure with no parameter), optional argument _generatorBrand_, and _generatorPrototype_ (an Object).</p>
+      <h1>CreateBuiltinGeneratorInstance ( _closure_ [ , _generatorBrand_ [ , _generatorPrototype_ [, _generatorSlots_] ] ] )</h1>
+      <p>The abstract operation CreateBuiltinGeneratorInstance takes arguments _closure_ (an Abstract Closure with no parameter), optional argument _generatorBrand_, _generatorPrototype_ (an Object) and _generatorSlots_ (a List of internal slots).</p>
       <emu-alg>
         1. Assert: _closure_ is an Abstract Closure. It can contain uses of the Yield shorthand to yield an IteratorResult object.
         1. If _generatorPrototype_ is not present, set _generatorPrototype_ to %Iterator.prototype%.
         1. If _generatorBrand_ is not present, set _generatorBrand_ to ~empty~.
-        1. Let _generator_ be ? OrdinaryObjectCreate(_generatorPrototype_, &laquo; [[GeneratorState]], [[GeneratorContext]], [[GeneratorBrand]] &raquo;).
+        1. If _generatorSlots_ is not present, set _generatorSlots_ to &laquo; &raquo;
+        1. Let _slots_ be concat &laquo; [[GeneratorState]], [[GeneratorContext]], [[GeneratorBrand]] &raquo; and _generatorSlots_.
+        1. Let _generator_ be ? OrdinaryObjectCreate(_generatorPrototype_, _slots_).
```

Comment:
To match what is done by [`OrdinaryObjectCreate`](https://tc39.es/ecma262/#sec-ordinaryobjectcreate), this should be:
```suggestion
        1. Let _internalSlotsList_ be concat &laquo; [[GeneratorState]], [[GeneratorContext]], [[GeneratorBrand]] &raquo;.
        1. If _generatorSlots_ is present, append each of its elements to _internalSlotsList_.
        1. Let _generator_ be ? OrdinaryObjectCreate(_generatorPrototype_, _internalSlotsList_).
```